docs/rules/better-boolean-variable-names.md

```diff
@@ -0,0 +1,162 @@
+# Prefer readable Boolean variable names
+
+💼 This rule is enabled in the ✅ `recommended` [config](https://github.com/sindresorhus/eslint-plugin-unicorn#preset-configs-eslintconfigjs).
+
+💡 This rule is manually fixable by [editor suggestions](https://eslint.org/docs/latest/use/core-concepts#rule-suggestions).
+
+<!-- end auto-generated rule header -->
+<!-- Do not manually modify this header. Run: `npm run fix:eslint-docs` -->
+
+When it is clear that an expression is a Boolean value, the variable name should start with `is`/`was`/`has`/`can`/`should` to improve readability.
+
+## Examples
+
+```js
+const completed = true; // ❌
+const isCompleted = true; // ✅
+```
+
+```js
+const completed = progress === 100; // ❌
+const isCompleted = progress === 100; // ✅
+```
+
+```js
+const completed = Boolean('true'); // ❌
+const isCompleted = Boolean('true'); // ✅
+```
+
+```js
+const completed = new Boolean('true'); // ❌
+const isCompleted = new Boolean('true'); // ✅
+```
+
+```js
+const adult = age >= 18; // ❌
+const isAdult = age >= 18; // ✅
+```
+
+```js
+const adult = age >= 18 ? true : false; // ❌
+const isAdult = age >= 18 ? true : false; // ✅
+```
+
+```js
+const gotModifyRights = isGotPreviewRights() && isGotDownloadRights(); // ❌
+const isGotModifyRights = isGotPreviewRights() && isGotDownloadRights(); // ✅
```

Comment:
Why? I think got is right. 

``` js
const gotMilk = true
```

Reply:
**Here is the response from AI:**

The word "got" is generally not suitable as a prefix for Boolean variables because it does not conform to the conventions for naming Boolean variables. Boolean variable names typically use verbs or adverbs to clearly express their Boolean value, such as "is," "has," "can," etc.

Using "got" as a prefix may result in variable names that are not sufficiently clear, thereby reducing the readability and maintainability of the code. Therefore, it is recommended to avoid using "got" as a prefix for Boolean variables.

Better choices include the following prefixes:

- `is`: isUserLoggedIn
- `has`: hasAdminPrivileges
- `can`: canEditProfile
- `should`: shouldDisplayNotification
- `will`: willSendEmail
- `needs`: needsPasswordReset

These prefixes can more clearly express the meaning of Boolean variables.